server/stream.go

```diff
@@ -2737,12 +2737,13 @@ func (mset *stream) setupMirrorConsumer() error {
 	} else {
 		mset.cancelSourceInfo(mset.mirror)
 		mset.mirror.sseq = mset.lseq
+	}
 
-		// If we are no longer the leader stop trying.
-		if !mset.isLeader() {
-			return nil
-		}
+	// If we are no longer the leader stop trying.
+	if !mset.isLeader() {
```

Comment:
De-flake for `TestJetStreamSuperClusterSourceAndMirrorConsumersLeaderChange` as it would still create a mirror consumer even if the stream was not a leader anymore after being stepped down.